crates/federated-server/src/server/mcp.rs

```diff
@@ -0,0 +1,40 @@
+mod server;
+
+use std::{
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4},
+    time::Duration,
+};
+
+use axum::Router;
+use engine::Runtime;
+use gateway_config::ModelControlProtocolConfig;
+use rmcp::transport::{SseServer, sse_server::SseServerConfig};
+use server::McpServer;
+use tokio_util::sync::CancellationToken;
+
+use super::gateway::EngineWatcher;
+
+pub(super) fn router<S, R: Runtime>(
```

Comment:
You don't need `S`, it expresses your router state _requirements_. So if you `Router` is self-sufficient it should return a `Router<()>`

Reply:
You just need to move the `merge` after the `.with_state()`, would avoid the generic here which is confusing imho